scripts/split-by-solvency.ts

```diff
@@ -0,0 +1,71 @@
+import { SuperTokenList, TokenInfo } from "..";
+import tokenList from "../superfluid.tokenlist.json";
+import fs from "fs";
+import { validate, validateUnderlyingTokens } from "../utils";
+
+const createFilteredList = async (tokenList: SuperTokenList, tier: string) => {
+  const underlyingTokens = tokenList.tokens.filter((token) =>
+    token.tags?.includes("underlying")
+  );
+
+  const filteredByTier = tokenList.tokens.filter((token) =>
+    token.tags?.includes(tier)
+  );
+
+  const tokens: TokenInfo[] = [];
+
+  filteredByTier.forEach((token) => {
+    tokens.push(token);
+
+    if (
+      ["Wrapper", "Native Asset"].includes(
+        token.extensions?.superTokenInfo?.type ?? ""
+      )
+    ) {
+      const underlyingToken = underlyingTokens.find(
+        (underlyingToken) =>
+          underlyingToken.address ===
+          // @ts-ignore (Special Native Assets have underlying tokens)
+          token.extensions?.superTokenInfo?.underlyingTokenAddress
+      );
+      if (underlyingToken) {
+        tokens.push(underlyingToken);
```

Comment:
Technically there is some small chance for duplicate underlying tokens getting added. It might be validated later in the process though.

Reply:
modified the find, to also include chainId